### DISCUSSION TITLE
Decouple module solution from oqtane.framework Git

### DISCUSSION THREAD

**Discussion by mdmontesinos** [oqtane/oqtane.framework#2324] [Q&A] [answered]
I've installed Oqtane by cloning the dev branch in Visual Studio but now the issue is that I can't have a separate Git repo for each module.

When creating a module or theme via the creator in Oqtane, it includes the reference to Oqtane.Server, as expected, but the solution for the module now gets associated with the Git repo for oqtane.framework and it does not even display changes made to the module itself. 

As an example, this solution:
![image](https://user-images.githubusercontent.com/90258222/182129297-2d844572-f358-4154-8039-13bbfcecd95f.png)
Has the following Git version control:
![image](https://user-images.githubusercontent.com/90258222/182129387-7f5165be-0df1-4078-964b-715c0e6316e6.png)

I would like to be able to have a separate and individual GitHub repository for each module or theme that I create.

I'm aware that this is not an issue related with Oqtane itself, but perhaps someone here can help me out understanding how Visual Studio handles Git repositories.

Thanks in advance!

**Comment by leigh-pointer** [comment, marked as answer]
@mdmontesinos remove the oqtane server save the solution add to github. Once it is added to github add the server reference again.

**Reply by mdmontesinos** [reply]
It worked. Thanks a lot!

**Comment by sbwalker** [comment]
@leigh-pointer I have actually had a lot of difficulty with the Oqtane.Server being included in the module solution - it can cause strange compilation errors, etc.. so I end up removing it after I create a new module. So although it is helpful in some scenarios, I am actually wondering if it should be removed from the template.

**Reply by leigh-pointer** [reply]
@sbwalker There shouldn't be any compilations errors as the Server Project should never be compiled.  I will explorer for a better solution but using the server reference is a lot fast than attaching processes.
